### PR TITLE
[GH-5951] Update Authentication > OAuth 2.0 > Office 365

### PR DESCRIPTION
**Pull request by matthew-src**
#### Summary

This PR updates the Office 365 section of the OAuth 2.0 settings to the table-based format. This completes the OAuth 2.0 section.

#### Ticket Link

This PR partly addresses https://github.com/mattermost/docs/issues/5951.

### CODE REVIEW

**Comment by cwarnermm** [comment]
Thanks for your patience on this PR, @matthew-src! 

Why there are conflicts on this PR
----------------------------------

You'll see merge conflicts for this PR due to https://github.com/mattermost/docs/pull/5911 being merged. The conflicts exist because PR-5911 was based off of what's in the master branch, and this PR is where your content updates are being applied. When GitHub detects a discrepancy between PRs, it provides you with both copies of text, and you remove the content that is no longer relevant. In most, if not all cases, you'll elect to keep your content updates, such as section heading updates and content clarifications.

What else has been added to this PR?
------------------------------------

When reviewing the conflicts in this PR, you'll also see new metadata specified for each config setting in the form of a directive and its options. This metadata is positioned directly above each section header on each config setting docs page, and will look something like this:

```rst
.. config:setting:: web-useletsencrypt
  :displayname: Use Let's Encrypt (Web Server)
  :systemconsole: Environment > Web Server
  :configjson: .ServiceSettings.UseLetsEncrypt
  :environment: MM_SERVICESETTINGS_USELETSENCRYPT
  :description: Enable the automatic retrieval of certificates from Let’s Encrypt.
```

The new setting directive allows us to surface up each config setting as a search result. This functionality is live in production via https://docs.mattermost.com/. Feel free to try it out yourself by searching for a term such as "timeout" or "ip". You'll see that config settings are now prioritized and returned first in the search results, ranked by relevancy. 

- The directive itself needs a unique identifier, such as ``web-useletsencrypt``. The directive's options, including ``displayname``, ``systemconsole``, ``configjson``, ``environment``, and ``description`` essentially duplicates the config setting details. We're doing this so that the config setting details readers want show up on the search results page.

The setting directive supports the following options:

``displayname``: User-friendly text that is displayed as the name of a config setting
``systemconsole``: The location in the System Console of the config setting
``configjson``: The config.json name of the config setting
``environment``: The environment variable for the config setting
``description``: Description of the config setting; can be any length

Next steps
----------

You shouldn't have to modify any of the metadata in this PR. However, if you do see a discrepancy between the content on the docs page itself versus what's populated in the metadata, please flag it for review and let me know. I'm happy to help identify the right content needed if there is a discrepancy.

Manage and resolve the merge conflicts
---------------------------------------

The trickiest part to managing and resolving merge conflicts in GitHub is ensuring that both the unwanted text and GitHub's text-based flags are removed. And knowing what text is safe to remove and what text should stay. It gets easier with practice :)

Working from the following example in this PR:
image

The first section I've highlighted is your new content:
image

And this second highlighted section is what was already in the master branch:
image

To resolve this specific merge conflict, you'd remove the first GitHub flag:
image

Then you'd remove the content you no longer want to keep from the master branch:
image

Once all individual conflicts are resolved for a given source file, GitHub allows you to mark the file as resolved at the top of the page. You would repeat these steps for each conflict in each impacted file. And you'd repeat this process for each of your in-flight PRs with merge conflicts, including this one and https://github.com/mattermost/docs/pull/6132. On first glance, https://github.com/mattermost/docs/pull/6134/files doesn't appear to have any conflicts which is great!

Let me know if any of this doesn't make sense, and if you'd like any assistance addressing the merge conflicts. Happy to help!

**Comment by matthew-src** [comment]
Thank you, @cwarnermm. I think I have successfully resolved the merge conflicts. I would like to flag the metadata descriptions for review in these settings, as I think the new description is more informative, and conforms to other documentation:

- Office 365 Application ID
- Office 365 Application secret password
- Office 365 Directory ID
- Office 365 OAuth 2.0 Auth endpoint

**Comment by cwarnermm** [comment]
Thanks, @matthew-src - I agree; please apply these changes. This is exactly the level and detail of review I was hoping for. Well done!

File: source/configure/authentication-configuration-settings.rst [hunk 1]
```diff
@@ -2036,69 +2036,72 @@ Office 365 OAuth 2.0 settings
...
-
-Application ID
-''''''''''''''
+  :description: This setting holds the Application ID generated when configuring Office 365 as a Single Sign-On service through the Microsoft Azure Portal.
 
 *Available in legacy Enterprise Edition E20*
 
-Obtain this value by registering Mattermost as an application in your Microsoft or Office account.
++-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------+
+| This setting holds the **Application ID** generated when configuring Office 365 as a Single Sign-On service through the Microsoft Azure Portal. See `Register an application in Azure Portal <https://docs.mattermost.com/onboard/sso-office.html#step-1-register-an-application-in-azure-portal>`__. | - System Config path: **Authentication > OAuth 2.0** |
```
Comment (cwarnermm) [inline source/configure/authentication-configuration-settings.rst:2062]: See my comment on https://github.com/mattermost/docs/pull/6172#discussion_r1085382223 about the format used to provide docs references and the feedback via https://github.com/mattermost/docs/pull/6132#issuecomment-1402051426 about potentially moving all docs references to a subsequent row for consistency.

**Review by cwarnermm — COMMENTED** [review]
Thanks, @matthew-src! Great job on these PRs. I've provided feedback inline for consistency.

**Comment by matthew-src** [comment]
Hi @cwarnermm, please see the updates in the latest commits. Should the URL endpoints described in the Office 365 OAuth settings be formatted as code or in bold? Right now they are formatted as code, and the setting description does reference the ``config.json`` file.

**Comment by cwarnermm** [comment]
Thanks, @matthew-src - I agree with your formatting decision. Well done!

**Review by cwarnermm — APPROVED** [review]

**Pull request merged**